src/webapp/components/survey-list/hook/useDeleteSurvey.ts

```diff
@@ -0,0 +1,39 @@
+import { Dispatch, SetStateAction, useState } from "react";
+import { SURVEY_FORM_TYPES } from "../../../../domain/entities/Survey";
+import { useCurrentSurveys } from "../../../contexts/current-surveys-context";
+
+import { useAppContext } from "../../../contexts/app-context";
+import i18n from "@eyeseetea/feedback-component/locales";
+import { Id } from "../../../../domain/entities/Ref";
+import { ActionOutcome } from "../../../../domain/entities/generic/ActionOutcome";
+
+export function useDeleteSurvey(
+    formType: SURVEY_FORM_TYPES,
+    refreshSurveys: Dispatch<SetStateAction<{}>>
+) {
+    const { compositionRoot } = useAppContext();
+    const [deleteCompleteState, setDeleteCompleteState] = useState<ActionOutcome>();
+    const { currentHospitalForm } = useCurrentSurveys();
```

Comment:
@mariaozamiz  could you add a loader when the delete is happening, so the user knows something is happening. You can refer to loading state in useSurveyForm.